zoltar-core/src/main/java/com/spotify/zoltar/fs/FileSystemExtras.java

```diff
@@ -40,6 +42,21 @@
   private FileSystemExtras() {
   }
 
+
+  /**
+   * Finds the latest date directory in the directory src. It assumes that partitions are
+   * formatted using ISO_LOCAL_DATE (e.g. YYYY-MM-DD).
+   */
+  public static String getLatestDate(String src) throws IOException {
+    return Files.list(Paths.get(src))
+        .filter(p -> Files.isDirectory(p))
+        .map(p -> LocalDate.parse(p.getFileName().toString()))
+        .sorted(Comparator.reverseOrder())
+        .map(d -> d.toString())
+        .collect(Collectors.toList())
+        .get(0);
```

Comment:
would it be better to use `.findFirst()` and return `Optional<String>`? if `Stream` is empty this will throw `ArrayIndexOutOfBoundsException` right?

Reply:
We can replace `sorted/map/collect/get` with `.reduce((x, y) -> x.compareTo(y) > 0 ? x : y)`

Reply:
@nevillelyh that's nicer